supervisor/worker.go

```diff
@@ -65,6 +65,13 @@ func (w *worker) Start() {
 		}
 		if err := process.Start(); err != nil {
 			logrus.WithField("error", err).Error("containerd: start init process")
+			t.Err <- err
```

Comment:
Shouldn't we do the same if monitor process fails? No way to get notified of container death  otherwise. 
Also, if the start fails, we should probably kill the init process as it'll stay stuck forever and there's no mechanism to trigger a start again (aside from doing it manually).